citesphere/src/test/java/edu/asu/diging/citesphere/web/validation/CitationConceptValidatorTest.java

```diff
@@ -0,0 +1,146 @@
+import edu.asu.diging.citesphere.model.IUser;
+import edu.asu.diging.citesphere.model.bib.ICitationConcept;
+import edu.asu.diging.citesphere.model.bib.impl.CitationConcept;
+import edu.asu.diging.citesphere.model.impl.User;
+import edu.asu.diging.citesphere.web.forms.CitationConceptForm;
+
+public class CitationConceptValidatorTest {
+
+    @InjectMocks
+    private CitationConceptValidator citationConceptValidator;
+
+    @Mock
+    private ICitationConceptManager conceptManager;
+
+    @Mock
+    private Authentication authentication;
+
+    @Mock
+    private SecurityContextHolder securityContextHolder;
+
+    @Mock
+    private SecurityContext securityContext;
+
+    @Mock
+    private ICitationConcept concept;
+
+    private IUser user;
+
+    private Errors errors;
+
+    private CitationConceptForm conceptForm;
+
+    private ICitationConcept dbConcept;
+
+    @Before
+    public void setUp() {
+
+        MockitoAnnotations.initMocks(this);
+
+        user = new User();
+        user.setUsername("akhil");
+
+        initForm();
+        initExistingConcept();
+        Mockito.when(securityContext.getAuthentication()).thenReturn(authentication);
+        Mockito.when(authentication.getPrincipal()).thenReturn(user);
+        SecurityContextHolder.setContext(securityContext);
+
+    }
+
+    private void initForm() {
+        conceptForm = new CitationConceptForm();
+        conceptForm.setDescription("test dec");
+        conceptForm.setName("name");
+        conceptForm.setUri("http://wwww.google.com");
+
+    }
+
+    private void initExistingConcept() {
+        dbConcept = new CitationConcept();
+        dbConcept.setOwner(user);
+        dbConcept.setId("CON1");
+        dbConcept.setUri("http://www.google.com");
+    }
+
+    @Test
+    public void validateTest_AddConcept_Success() {
+
+        // return null for adding concept
+        Mockito.when(conceptManager.getByUriAndOwner("http://wwww.google.com", user)).thenReturn(null);
+        errors = new BeanPropertyBindingResult(conceptForm, "invalidConcept");
+        citationConceptValidator.validate(conceptForm, errors);
+        assert (errors.getFieldError("uri") == null);
+
+    }
+
+    @Test
+    public void validateTest_AddConcept_Uri_Exists() {
+
+        Mockito.when(conceptManager.getByUriAndOwner("http://wwww.google.com", user)).thenReturn(concept);
+        errors = new BeanPropertyBindingResult(conceptForm, "validConcept");
+        citationConceptValidator.validate(conceptForm, errors);
+        assert (errors.getFieldError("uri") != null);
+
+    }
+
+    @Test
+    public void validateTest_EditConcept_Success() {
+
+        conceptForm.setConceptId("CON1");
+        dbConcept.setUri("http://www.google1.com");
+        Mockito.when(conceptManager.get("CON1")).thenReturn(dbConcept);
+        Mockito.when(conceptManager.getByUriAndOwner("http://wwww.google.com", user)).thenReturn(null);
+        errors = new BeanPropertyBindingResult(conceptForm, "validConcept");
```

Comment:
for line 111: the uri you set in the concept is `http://www.google1.com`, why do you mock with uri `http://wwww.google.com`?

Reply:
`http://www.google1.com` is the original Uri in the database. 
`http://wwww.google.com` is the uri which is the new uri entered through conceptForm.

Reply:
that makes sense; I think I need more coffee. let's make all of these constants with telling names though.

Reply:
Instead of using String literal,  I am taking uri from the conceptForm now.